web/src/main/java/com/epam/esm/controller/AuthController.java

```diff
@@ -0,0 +1,31 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.dto.TokenRequest;
+import com.epam.esm.dto.UserRegisterDTO;
+import com.epam.esm.service.UserService;
+import jakarta.validation.Valid;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/api/auth", produces = "application/json", consumes = "application/json")
+public class AuthController {
+    private final UserService userService;
+
+    public AuthController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @PostMapping("/generate-token")
+    public ResponseEntity<String> generateToken(@Valid @RequestBody TokenRequest request) {
```

Comment:
Maybe it would be better if this endpoint is called as **login** not **generate-token**